lib/ipmi_backend_utils.pm

```diff
@@ -528,4 +528,21 @@ sub set_grub_terminal_and_timeout {
     }
 }
 
+sub reconnect_when_ssh_console_broken {
+
+    # Switch to sol console to check serial console output
+    # It is useful in the case of host crash and reboot
+    record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'fail');
+    select_console 'sol', await_console => 0;
+    save_screenshot;
+    # Wait host bootup if it crashes
+    die "Unable to connect machine" unless check_port_state(get_required_var('SUT_IP'), 22, 10);
```

Comment:
```
if check port is true {
  xxxxx
}
```
You are already in post_fail_hook. I am afraid using "die" here will discontinue current flow and redirect to somewhere else, for example, parent functions ?

Reply:
oh.... If host is not up, then there is no need to continue anymore ? Verify your test flow anyway.